File: LeetCode/Easy/1957-Delete_Characters_To_Make_Fancy_String/1957-DeleteCharactersToMakeFancyString.cpp
#include <string>

class Solution
{
public:
    string makeFancyString(string s)
    {
        if (s.size() < 3)
            return s;

        char prev = s[0];
        int count = 1;
        string ans = "";
        ans.push_back(s[0]);

        for (int i = 1; i < s.size(); i++)
        {
            if (prev == s[i])
                count++;
            else
                count = 1;

            if (count < 3)
                ans.push_back(s[i]);

            prev = s[i];
        }

        return ans;
    }
};

File: LeetCode/Easy/1957-Delete_Characters_To_Make_Fancy_String/1957-DeleteCharactersToMakeFancyString.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        if len(s) < 3:
            return s

        prev = s[0]
        count = 1
        ans = [s[0]]

        for c in s[1:]:
            if prev == c:
                count += 1
            else:
                count = 1

            if count < 3:
                ans.append(c)

            prev = c

        return "".join(ans)


a = Solution()
print(a.makeFancyString(s="leeetcode"))
print(a.makeFancyString(s="aaabaaaa"))
print(a.makeFancyString(s="aab"))
